lmfit/minimizer.py

```diff
@@ -1258,7 +1260,8 @@ def _lnpost(theta, userfcn, params, var_names, bounds, userargs=(),
 
     # now calculate the log-likelihood
     out = userfcn(params, *userargs, **userkwargs)
-    out = _nan_policy(out, nan_policy=nan_policy, handle_inf=False)
```

Comment:
This comment applies here, and on L363. `_nan_policy` was designed in such a way to be really efficient. If out is a large array (it's not unusual to have many datapoints) then `np.isnan(out)` creates a large temporary `bool` array (with associated overhead). `_nan_policy` circumvents this by a reduction operation first, `np.sum`, and checking if that single value is finite.
I had thought of implementing the checks in this commit, but I think the code looks neater as it exists currently, and I don't think there is any real benefit (no extra overhead) to making these two changes.